src/registry.js

```diff
@@ -209,8 +209,16 @@ exports.getComponentMethod = function(name, method) {
 };
 
 /**
+ * Call registered api method.
  *
+ * @param {string} name : api method name
+ * @param {array} args : array of argument passed to api method
+ * @return {any} : returns api method output
  */
+exports.call = function(name, args) {
+    return exports.apiMethodRegistry[name].apply(null, args);
+};
```

Comment:
Can we get rid of the array wrapping the args, like other `call` methods (`d3.select().call` and native `function.call`)? Borrowing from d3 and condensing, seems like it would just be:
```
return exports.apiMethodRegistry[name].apply(null, [].slice.call(arguments, 1));
```
or, perhaps marginally faster if we don't care about `this` being the method name:
```
return exports.apiMethodRegistry[name].apply([].slice.call(arguments));
```

Reply:
That's what I was going to do until I saw this [block](https://github.com/plotly/plotly.js/blob/113b3618732f28d1a2158771544bf4e60e00c084/src/plots/command.js#L264-L281) (called from updatemenus and silders) where the argument list is built as an array.

I think I would prefer keeping this pattern. I could rename `Registry.call` -> `Registry.apply` to avoid confusion on the call signature if you prefer.

Reply:
Ah right, that's the one place you use `Registry.call` without brackets, and it would be awkward to try and fit that into the `call` pattern. It's a shame that this one call adds cruft to the hundred other ones though. What about creating both `Registry.call` *and* `Registry.apply` methods? Or, since at least for now that's the only place we'd use `Registry.apply`, just inlining it there:
```
return Registry.apiMethodRegistry[method].apply(null, allArgs).catch(...
```

Reply:
Done in https://github.com/plotly/plotly.js/pull/2429/commits/705aed622d42648d55b4497c93486ea96921e1a0 :rocket: